### PR TITLE
docs: update workarounds for sub-components with Next.js App router

### PR DESCRIPTION
**Pull request by navidmafi**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
workaround for https://github.com/vercel/next.js/issues/51593

### 💡 Background and solution

Using subcomponents of components (like `Typography` in Next.js app router causes an error that tells that client components are not dottable on the server.
The current proposed fix is just to make the page fully client-rendered which works well as a workaround but is less than optimal since by making the component fully client-rendered developers can't use SSR patterns if they want to.

The new workaround that I'm proposing here is to extract the needed subcomponents into client components which allows for usage is SSR pages.  

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: docs/react/faq.en-US.md [hunk 1]
```diff
@@ -268,13 +268,48 @@ The above problem occurs if `strictNullChecks` is set to `true`, If you can dete
...
 
-If you are using the App Router of Next.js, when you use the sub-components provided by some antd components, such as `Select.Option `, `Form.Item`, etc., you may get the following error:
+If you are using the App Router of Next.js, when you use the sub-components provided by some antd components, such as `Select.Option `, `Form.Item`, `Typography.Title`, etc., you may get the following error:
 
 ```bash
 Error: Cannot access .Option on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
 ```
 
-At present, this problem is waiting for Next.js to give an official solution, before this, if you use sub-components in your page, you can try to add the following client tag at the top of the page to solve this problem:
+Currently, this problem is [waiting for Next.js to give an official solution](https://github.com/vercel/next.js/issues/51593). There are two workarounds as of now if you need to use sub-components in your page with the App Router:
```
Comment (yoyo837) [inline docs/react/faq.en-US.md:277]: Please help to add Chinese documents changes also.

Comment (navidmafi) [reply docs/react/faq.en-US.md:277]: Unfortunately, I can't speak Chinese, and the best I could do it to use Google Translate, DeepL machine translation.

Is that acceptable?

Comment (yoyo837) [reply docs/react/faq.en-US.md:277]: Yes, and we can suggest you after you add that changes.

**Comment by navidmafi** [comment]
Added zh-CN translation using Machine translation (DeepL). Please let me know if any part is wrongly translated.

File: docs/react/faq.zh-CN.md [hunk 2]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
 
-如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item` 等，可能会出现如下报错：
+如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item`、`Typography.Title` 等，可能会出现如下报错：
 
 ```bash
 Error: Cannot access .Option on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
 ```
 
-目前这个问题等待 Next.js 给出官方的解决方案，在此之前，如果在你的页面中有使用子组件的话，可以尝试在页面顶部增加如下客户端标签解决这个问题：
+[目前这个问题等待 Next.js 给出官方的解决方案](https://github.com/vercel/next.js/issues/51593). 如果您需要在使用应用程序路由器的页面中使用子组件，目前有两种变通方法：
```
Comment (yoyo837) [inline docs/react/faq.zh-CN.md:306]:
```suggestion
目前这个问题[等待 Next.js 给出官方的解决方案](https://github.com/vercel/next.js/issues/51593). 如果您需要在使用应用程序路由器的页面中使用子组件，目前有两种变通方法：
```

File: docs/react/faq.zh-CN.md [hunk 3]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
+
+- 创建一个封装组件，提取所需的子组件并重新导出。以排版组件为例。包装组件的外观是这样的
```
Comment (yoyo837) [inline docs/react/faq.zh-CN.md:308]:
```suggestion
- 创建一个包装组件，提取所需的子组件并重新导出。以 `Typography` 组件为例。包装组件的代码大概像这样
```

File: docs/react/faq.zh-CN.md [hunk 4]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
+export { Title, Link, Text, Paragraph };
+```
+
+- 您也可以在页面源代码的开头添加 "use client "标签，使页面完全由客户端渲染：
 
 ```tsx
 'use client';
 
-// This is not real world code, just for explain
+// 非真实代码，仅供解释
```
Comment (yoyo837) [inline docs/react/faq.zh-CN.md:345]:
```suggestion
// 非真实代码，仅供逻辑说明
```

Comment (yoyo837) [inline docs/react/faq.zh-CN.md:345]:
```suggestion
// 非真实代码，仅做逻辑说明
```

File: docs/react/faq.zh-CN.md [hunk 5]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
 
-如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item` 等，可能会出现如下报错：
+如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item`、`Typography.Title` 等，可能会出现如下报错：
 
 ```bash
 Error: Cannot access .Option on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
 ```
 
-目前这个问题等待 Next.js 给出官方的解决方案，在此之前，如果在你的页面中有使用子组件的话，可以尝试在页面顶部增加如下客户端标签解决这个问题：
+目前这个问题[等待 Next.js 给出官方的解决方案](https://github.com/vercel/next.js/issues/51593). 如果您需要在使用应用程序路由器的页面中使用子组件，目前有两种变通方法：
```
Comment (li-jia-nan) [inline docs/react/faq.zh-CN.md:306]:
```suggestion
目前这个问题需要[等待 Next.js 官方给出解决方案](https://github.com/vercel/next.js/issues/51593)，如果你需要在使用 App router 的页面中使用子组件，目前有两种变通方法：
```

File: docs/react/faq.zh-CN.md [hunk 6]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
+如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item`、`Typography.Title` 等，可能会出现如下报错：
 
 ```bash
 Error: Cannot access .Option on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
 ```
 
-目前这个问题等待 Next.js 给出官方的解决方案，在此之前，如果在你的页面中有使用子组件的话，可以尝试在页面顶部增加如下客户端标签解决这个问题：
+目前这个问题需要[等待 Next.js 官方给出解决方案](https://github.com/vercel/next.js/issues/51593)，如果你需要在使用 App router 的页面中使用子组件，目前有两种变通方法：
+
+- 创建一个包装组件，提取所需的子组件并重新导出。以 `Typography` 组件为例。包装组件的代码大概像这样
```
Comment (li-jia-nan) [inline docs/react/faq.zh-CN.md:308]:
```suggestion
- 创建一个包裹组件，提取所需的子组件并重新导出。以 `Typography` 组件为例，代码大概像这样：
```

File: docs/react/faq.zh-CN.md [hunk 7]
```diff
@@ -297,18 +297,52 @@ export default () => {
...
+Link.displayName = "ClientLink";
+
+const Text: ForwardRefExoticComponent<TextProps & RefAttributes<HTMLElement>> =
+  forwardRef((props, ref) => <_Typography.Text ref={ref} {...props} />);
+Text.displayName = "ClientText";
+
+export { Title, Link, Text, Paragraph };
+```
+
+- 您也可以在页面源代码的开头添加 "use client "标签，使页面完全由客户端渲染：
```
Comment (li-jia-nan) [inline docs/react/faq.zh-CN.md:340]:
```suggestion
- 你也可以在组件的开头添加 "use client" 指令，使页面完全由客户端渲染：
```

File: docs/react/faq.zh-CN.md [hunk 8]
```diff
@@ -297,18 +297,51 @@ export default () => {
...
 
-如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item` 等，可能会出现如下报错：
+如果你在使用 Next.js 的 App Router，当你使用 antd 中某些组件提供的子组件，如：`Select.Option`、`Form.Item`、`Typography.Title` 等，可能会出现如下报错：
 
 ```bash
 Error: Cannot access .Option on the server. You cannot dot into a client module from a server component. You can only pass the imported name through.
 ```
 
-目前这个问题等待 Next.js 给出官方的解决方案，在此之前，如果在你的页面中有使用子组件的话，可以尝试在页面顶部增加如下客户端标签解决这个问题：
+目前这个问题需要[等待 Next.js 官方给出解决方案](https://github.com/vercel/next.js/issues/51593)，如果你需要在使用 App router 的页面中使用子组件，目前有两种变通方法：
```
Comment (li-jia-nan) [inline docs/react/faq.zh-CN.md:306]:
```suggestion
目前这个问题需要[等待 Next.js 官方给出解决方案](https://github.com/vercel/next.js/issues/51593)，在此之前，如果你需要在使用 App router 的页面中使用子组件，目前有两种变通方法：
```

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**